stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/composer/MessageComposerController.kt

```diff
@@ -243,4 +292,56 @@ public class MessageComposerController(
         setMessageMode(MessageMode.Normal)
         dismissMessageActions()
     }
+
+    /**
+     * Cancels any pending work when the parent ViewModel is about to be destroyed.
+     */
+    public fun onCleared() {
+        scope.cancel()
+    }
+
+    /**
+     * Checks the current input for validation errors.
+     */
+    private fun validateInput() {
+        validationErrors.value = mutableListOf<ValidationError>().apply {
+            val messageLength = input.value.length
+            if (messageLength > maxMessageLength) {
+                add(
+                    ValidationError.MessageLengthExceeded(
+                        messageLength = messageLength,
+                        maxMessageLength = maxMessageLength
+                    )
+                )
+            }
+
+            val attachmentCount = selectedAttachments.value.size
+            if (attachmentCount > maxAttachmentCount) {
+                add(
+                    ValidationError.AttachmentCountExceeded(
+                        attachmentCount = attachmentCount,
+                        maxAttachmentCount = maxAttachmentCount
+                    )
+                )
+            }
+
+            val attachments: List<Attachment> = selectedAttachments.value
+                .filter { it.fileSize > maxAttachmentSize }
```

Comment:
One thing that we need to note here - I think the max attachment size od 5MB is a BE limitation. So we should make the check that the file size also isn't larger than that.

Something like:

`it.fileSize > maxAttachmentSize || it.fileSize > MAX_DEFAULT_FILE_SIZE`. It kind of seems ugly but it's the way things should work

Reply:
Might be the same for number of attachments - as I'm not sure that messages can have more than 10 attachments

Reply:
So, the following are the backend limitations:
- 20mb for Stream CDN, any file size for a custom CDN.
- 10 attachments per message. This looks like a value hardcoded on the backend, I can limit enforce the upper limit to 10 items. Don't want to crash if a client specifies something like 20 attachments per message.
- 5000 chars per message, but can be set to a bigger value via dashboard.

Reply:
This all sounds good and if we can observe things here using channel config, that's great. If not, we can put these default limits and add support for dashboard customization later on!